### DISCUSSION TITLE
Block or allow websites based on keywords on the title of the page

### DISCUSSION THREAD

**Discussion by karthikeyann** [proginosko/LeechBlockNG#359] [Ideas]
Right now, the rules apply to the URL only. 
Request for feature to block or allow websites based on keywords on the title of the page.

For example:
The following URL does not have CppCon or C++ in the URL.
https://www.youtube.com/watch?v=zL-vn_pGGgY
But the page title contains keywords like "C++" and "CppCon". 
Leechblock allows only to [block/override based on keywords in the URL](https://www.proginosko.com/leechblock/faq/keywords/) right now (as far as i know).

Is it possible to add the feature to overrides exceptions based on keywords in the title of the page?

**Comment by alerezaaa** [comment]
Hi
please check the last paragraph: https://www.proginosko.com/leechblock/faq/keywords/
> If you want pages with the specified keywords to be allowed rather than blocked, you should select the option Treat keywords as allow-conditions rather than block-conditions under Advanced Options.

**Reply by karthikeyann** [reply]
I tried for "C++" (in the content of the page). I couldn't get the escaping `+` character right. It didn't work with `~C++` or `~C\+\+`.

**Reply by proginosko** [reply]
Yeah, currently keywords can only be alphanumerical (with underscores for spaces).

**Comment by proginosko** [comment]
The keyword blocking feature currently applies only to the **text content** of the page. But it would be easy to include the title text in the search. Look out for it in a future release.

**Comment by forestlink366** [comment]
I am attempting to do the same thing as OP - block all of youtube except a handful of videos from a select few channels. Like OP, this is challenging because of the random number string in the web address that doesn't have any useful keywords for the video. Looking at the post linked by @alerezaaa, allowing based on text in the page's body text seems possible, but I have not been able to impliment successfully. Not sure if user error or if I need to wait for @proginosko to add the title update.

For example, I tried things like adding ~@MandarinCorner2 ; ~chinese; and ~corner to whitelist [videos](https://www.youtube.com/watch?v=9em4FkmfMS8) with the "Treat keywords as allow-conditions rather than block-conditions" option both enabled and disabled, but to no success. In one instance I could get the @MandarinCorner2 channel page to be whitelisted, but that doesn't help me if I cannot go to the videos themselves.

**Reply by alerezaaa** [reply]
Hi
take a look at https://github.com/proginosko/LeechBlockNG/issues/391

**Comment by proginosko** [comment]
Implemented in version 1.6.3.
